src/vm/analysis/read_only_checker/mod.rs

```diff
@@ -290,27 +341,31 @@ impl<'a, 'b> ReadOnlyChecker<'a, 'b> {
                     _ => return Err(CheckError::new(CheckErrors::ContractCallExpectName)),
                 };
 
-                self.check_all_read_only(&args[2..])
+                self.check_each_expression_is_read_only(&args[2..])
                     .map(|args_read_only| args_read_only && is_function_read_only)
             }
         }
     }
 
-    fn check_function_application_read_only(
+    /// Checks the native function application implied by `expressions`. The first
```

Comment:
This checks any function application -- not _just_ native function applications (e.g. if `expressions` is a user-defined-function application).

Reply:
done